### PR TITLE
Query Prefetch OrderedMerge

### PR DESCRIPTION
**Pull request by tvaron3**
# Description

During an orderby query, more pages than necessary were being prefetched causing extra roundtrips in some cases. The default prefetch would grab 256 pages. It was changed to 1, so now the expected behavior is that for each physical partition one page will be prefetched.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/OrderByUtils.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public static <T extends Resource> Flux<OrderByRowResult<Document>> orderedMerge
                                                        targetRangeToOrderByContinuationTokenMap,
                                                        consumeComparer.getSortOrders(), clientSideRequestStatistics))
                 .toArray(Flux[]::new);
-        return Flux.mergeOrdered(consumeComparer, fluxes);
+        return Flux.mergeOrdered(1, consumeComparer, fluxes);
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/OrderByUtils.java:52]: Worth adding a comment here to explain why we are prefetching just 1. Similar to the comment we added on the flatMap prefetch fix PR.

**Review by kushagraThapar — COMMENTED** [review]
Missing changelog.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/OrderByUtils.java (hunk 1 above)
Comment (tvaron3) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/OrderByUtils.java:53]: Added the comment

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/OrderbyDocumentQueryTest.java [hunk 2]
```diff
@@ -159,6 +163,22 @@ public void queryOrderBy(String sortOrder) throws Exception {
...
     }
 
+    @Test(groups = { "query" }, timeOut = TIMEOUT)
+    public void queryOrderByRoundTrips() {
+        String query = "SELECT c.v FROM c where c.type='testing' order by c.v asc";
+        CosmosQueryRequestOptions options = new CosmosQueryRequestOptions();
+
+        int pageSize = 5;
+        CosmosPagedFlux<RoundTripDocument> queryObservable = roundTripsContainer.queryItems(query, options, RoundTripDocument.class);
+        FeedResponse<RoundTripDocument> feedResponse = queryObservable.byPage(pageSize).take(1).blockFirst();
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/OrderbyDocumentQueryTest.java:173]: how do we verify there is only 1 query request being sent to the backend?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/OrderByUtils.java [hunk 3]
```diff
@@ -49,7 +49,8 @@ public static <T extends Resource> Flux<OrderByRowResult<Document>> orderedMerge
                                                        targetRangeToOrderByContinuationTokenMap,
                                                        consumeComparer.getSortOrders(), clientSideRequestStatistics))
                 .toArray(Flux[]::new);
-        return Flux.mergeOrdered(consumeComparer, fluxes);
+        // prefetch is set to 1 to minimize the no. prefetched pages per partition
+        return Flux.mergeOrdered(1, consumeComparer, fluxes);
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/OrderByUtils.java:53]: Flux.mergeOrdered is going to be deprecated, maybe we can change to use the latest `mergeComparingDelayError`

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/OrderbyDocumentQueryTest.java (hunk 2 above)
Comment (tvaron3) [reply sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/OrderbyDocumentQueryTest.java]: True it is possible without the change that the test might pass. It should be max 4 requests for this test. One for each partition and one for each prefetch per partition. To make it more robust, I do it 100 times and I'm checking the outputDocuments to be max 10 (2 pages) and the requests captured in the diagnostics to be max 4. With these changes, the test will be unlikely to pass without the change.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/OrderByUtils.java (hunk 3 above)
Comment (tvaron3) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/OrderByUtils.java]: Made the change.

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

**Pull request merged**